pkg/scheduler/plugins/proportion/proportion.go

```diff
@@ -159,9 +159,11 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 
 			oldDeserved := attr.deserved.Clone()
 			attr.deserved.Add(remaining.Clone().Multi(float64(attr.weight) / float64(totalWeight)))
-
-			if attr.capability != nil && !attr.deserved.LessEqual(attr.capability, api.Infinity) {
+			if attr.capability != nil {
 				attr.deserved = helpers.Min(attr.deserved, attr.capability)
+			}
+
+			if attr.capability != nil && attr.capability.LessEqual(attr.deserved, api.Zero) {
```

Comment:
What if some dimensions are lack in some queue's capability? For example, capability of q1 = {cpu:10c}, but there must be `cpu` and `memory` in deserved.

Reply:
In the previous code, `helpers.Min` also make the deserved memory is 0.
``` go
if attr.capability != nil && !attr.deserved.LessEqual(attr.capability, api.Infinity) {
        attr.deserved = helpers.Min(attr.deserved, attr.capability)
……
```
In the current semantics, this situation means that the memory deserved is 0 always.
I think this pr didn't make it worse.

But if necessary, I can fix this problem in this pr or another pr soon.

Reply:
I get your meaning. In fact, deserved cpu and memory will never be zero for they are necessary dimensions. What I want to express is that lack dimension in capability should be in infinity.